orttraining/orttraining/python/training/ortmodule/_utils.py

```diff
@@ -38,21 +40,65 @@ def _ortvalue_from_torch_tensor(torch_tensor):
     return C.OrtValue.from_dlpack(to_dlpack(torch_tensor), is_bool_tensor)
 
 
-def _torch_tensor_from_dl_pack(dlpack, ortvalue):
-    torch_tensor = from_dlpack(dlpack)
-    return torch_tensor.to(torch.bool) if ortvalue.data_type() == 'tensor(bool)' else torch_tensor
+def _ortvalues_to_torch_tensor(ortvalues, device):
+    if len(ortvalues) == 0:
+        return tuple()
+    if hasattr(ortvalues, 'to_dlpacks'):
+        if all(ortvalue.proto_type() != TensorProto.BOOL for ortvalue in ortvalues):
```

Comment:
Is it possible to move all the BOOL handling code to the end of the function? For example, not return directly for each IF clause, but get a tensor_list without handling BOOL tensor. And at the end of this function, add:

return tuple([tensor.to(torch.bool) if ortvalues[i].proto_type() == TensorProto.BOOL else tensor for i, tensor in enumerate(tensor_list)])

With this way, we can also use the new to_dlpacks functions for a ortvalue list that also contains bool tensor, because it has better perf?